sdk/webpubsub/azure-messaging-webpubsub-client/src/main/java/com/azure/messaging/webpubsub/client/models/WebPubSubProtocol.java

```diff
@@ -0,0 +1,24 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.messaging.webpubsub.client.models;
+
+/**
+ * The superclass of protocol for WebPubSub client.
+ */
+public abstract class WebPubSubProtocol {
+
+    /**
+     * Gets whether the protocol is reliable.
+     *
+     * @return whether the protocol is reliable.
+     */
+    public abstract boolean isReliable();
+
+    /**
+     * Gets the name of the protocol.
+     *
+     * @return the name of the protocol.
+     */
+    public abstract String getName();
```

Comment:
As first beta, part of the logic about `isBinary`, `parseMessage`, `writeMessage` is coupled with client implementation and put in implementation package.

Will refactor them to `WebPubSubProtocol` abstract class before GA. However this would mean we also going to expose the full array of 'WebPubSubMessage' class to user (10+ of them).